### PR TITLE
Add task solution

### PR DESCRIPTION
**Pull request by YevhenOstrovskiy**
- [DEMO LINK](https://YevhenOstrovskiy.github.io/layout_product-cards/)
- [TEST REPORT LINK](https://YevhenOstrovskiy.github.io/layout_product-cards/report/html_report/)

- [x] there 2 BEM blocks `card` and `stars` each in it own file
- [x] SCSS Nesting is used for `elements`, `modifiers` and `pseudo-classes`
- [x] SCSS Variables are used for main values and placed in a **separate file**
- [x] all `stars--N` modifiers work as expected (Highlight first `N` stars)
- [x] Code follows all the [Code Style Rules ❗️](https://mate-academy.github.io/layout_task-guideline/html-css-code-style-rules)

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -8,6 +8,42 @@
     <link rel="stylesheet" href="./styles/index.scss">
   </head>
   <body>
-    <h1>Product cards</h1>
+    <div
+      class="card"
+      data-qa="card"
+    >
```
Comment (dinaminahi) [inline src/index.html:14]:
```suggestion
    <div class="card" data-qa="card">
```

We put attributes in new line if element has long attribute values or number of attributes is more than 2

File: src/index.html [hunk 2]
```diff
@@ -8,6 +8,42 @@
...
+          <div class="stars__star"></div>
+          <div class="stars__star"></div>
+          <div class="stars__star"></div>
+          <div class="stars__star"></div>
+          <div class="stars__star"></div>
+          <!-- #endregion -->
+        </div>
+        <span class="card__rating-reviews">Reviews: 5</span>
+      </div>
+
```
Comment (dinaminahi) [inline src/index.html:34]:
```suggestion
```

File: src/styles/style.css [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  background-repeat: no-repeat;
+  background-position: center;
+  margin-right: 4px;
+  height: 16px;
+  width: 16px;
+}
+
+.stars--4 .stars__star:nth-child(-n+4) {
+  background-image: url(../images/star-active.svg);
+  height: 16px;
```
Comment (dinaminahi) [inline src/styles/style.css:30]: we don't need it here anymore as you moved it to the card and stars scss files

File: src/styles/blocks/card.scss [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+  &__rating {
+    justify-content: space-between;
+    align-items: center;
+    width: 100%;
+    margin-bottom: 24px;
+    display: flex;
+
+    &-reviews {
+      color: #060B35;
```
Comment (dinaminahi) [inline src/styles/blocks/card.scss:45]: Let's put all colors to the variables

File: src/styles/index.scss [hunk 5]
```diff
@@ -1,3 +1,10 @@
 body {
   margin: 0;
+  font-family: Roboto, sans-serif;
 }
+
+@import 'https://fonts.googleapis.com/css2?family=Roboto:wght@100;400;500;700&display=swap';
+@import './blocks/variables';
+@import './blocks/card';
+@import './blocks/button';
+@import './blocks/stars';
```
Comment (dinaminahi) [inline src/styles/index.scss:10]:
```suggestion
@import 'https://fonts.googleapis.com/css2?family=Roboto:wght@100;400;500;700&display=swap';
@import './blocks/variables';
@import './blocks/card';
@import './blocks/button';
@import './blocks/stars';

body {
  margin: 0;
  font-family: Roboto, sans-serif;
}
```

Let`s firstly import files and then write your styles

File: src/styles/blocks/button.scss [hunk 6]
```diff
@@ -0,0 +1,22 @@
+.button {
+  box-sizing: border-box;
+  text-transform: uppercase;
+  color: $color-white;
+  background-color: $color-blue;
+  border: 1px solid $color-blue;
+  border-radius: 5px;
+  justify-content: center;
+  align-items: center;
+  width: 100%;
```
Comment (SerhiiKirik) [inline src/styles/blocks/button.scss:10]: Let's try to organize your CSS
I recommend you take a look at these styles code and use it in the feature
[9elements](https://9elements.com/css-rule-order) or [htmlAcademy](https://codeguide.academy/html-css.html#css-order)

File: src/styles/blocks/button.scss [hunk 7]
```diff
@@ -0,0 +1,22 @@
+.button {
+  box-sizing: border-box;
```
[the remark this answers is not in the document]
Comment (SerhiiKirik) [reply src/styles/blocks/button.scss:2]: Also, remove it in all cases

File: src/styles/index.scss [hunk 8]
```diff
@@ -1,3 +1,10 @@
...
+
 body {
   margin: 0;
+  font-family: Roboto, sans-serif;
```
Comment (SerhiiKirik) [inline src/styles/index.scss:9]: move it to the `html` selector
```suggestion
```

File: src/styles/blocks/card.scss [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+  &__image {
+    width: 160px;
+    height: 134px;
+    margin-bottom: 40px;
+    display: flex;
+  }
+
+  &__name {
+    color: $color-black-main;
+    margin-top: 0;
```
Comment (SerhiiKirik) [inline src/styles/blocks/card.scss:21]: remove it in `index.scss` file or create a separate `reset.scss` file for reset default rules

File: src/styles/blocks/card.scss [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+  align-items: center;
+  width: 200px;
+  padding:  32px 16px 16px;
+  display: flex;
+
+  &__image {
+    width: 160px;
+    height: 134px;
+    margin-bottom: 40px;
+    display: flex;
```
Comment (SerhiiKirik) [inline src/styles/blocks/card.scss:16]: mb block?
```suggestion
    display: block;
```

**Pull request closed without merging**